File: invertido.py
import cv2

fuente = cv2.imread('puppy.jpg')

#reducido = cv2.resize(img_fuente, (0,0), fx=0.25, fy=0.25)

vertical = cv2.flip(fuente, 0)
horizontal = cv2.flip(fuente, 1)
ambos = cv2.flip(fuente, -1)

cv2.imshow('original', fuente)
cv2.imshow('horizontal', horizontal)
cv2.imshow('vertical', vertical)
cv2.imshow('ambos', ambos)

cv2.waitKey(0)
cv2.destroyAllWindows()

File: cpp/invertido.cpp
#include <opencv2/opencv.hpp>
using namespace cv;

int main( int argc, char** argv )
{
	Mat origen;

	// Leer imagen
 	origen = imread("puppy.jpg", IMREAD_COLOR);

	if(!origen.data )
	{
   	printf( "No existe la imagen\n " );
   	return -1;
	}

	Mat vertical;
	Mat horizontal;
	Mat ambos;
	flip(origen, vertical, 0);
	flip(origen, horizontal, 1);
	flip(origen, ambos, -1);

	imshow( "Original", origen);
	imshow( "Vertical", vertical);
	imshow( "Horizontal", horizontal);
	imshow( "Ambos", ambos);

	waitKey(0);

	return 0;
}
